src/ws_server.ts

```diff
@@ -0,0 +1,126 @@
+/* This is free and unencumbered software released into the public domain. */
+
+import { Config } from './config.js';
+import { createServer } from './server.js';
+import pg from 'pg';
+import expressWs from 'express-ws';
+import { Engine, exportJSON } from '@aurora-is-near/engine';
+import { Logger } from 'pino';
+
+export function createWsServer(
+  config: Config,
+  logger: Logger,
+  engine: Engine,
+  app: any
+): boolean {
+  const pgClient = new pg.Client(config.database);
+  pgClient.connect();
+  const jaysonWsServer = createServer(config, logger, engine)
+  const expressWsApp = expressWs(app);
+  expressWsApp.app.ws("/", function(ws: any, req) {
+    ws.id = req.headers['sec-websocket-key'];
+    ws.on('message', function(msg: string) {
+      try {
+        const parsedObject = JSON.parse(msg);
+        jaysonWsServer.call(parsedObject, req, function (error: any, success: any) {
+          ws.send(JSON.stringify(error || success));
+        });
+      } catch(e) {
+        ws.send('Invalid request');
+      }
+    });
+    ws.on('close', function close() {
+      pgClient.query("DELETE FROM subscription WHERE sec_websocket_key = $1", [ws.id])
+    });
+    ws.on("error", (e: any) => ws.send(e));
+    ws.send('Connection established');
+  });
+
+  // Listen to new block notifications:
+  pgClient.on('notification', (message: pg.Notification) => {
+    if (!message.payload) return;
+    if (message.channel === 'block') {
+      const blockID = parseInt(message.payload);
+      if (isNaN(blockID)) return; // ignore UFOs
+      const body = {"jsonrpc": "2.0", "id": 1, "method": "eth_getBlockByNumber", "params": [blockID, true] }
+      jaysonWsServer.call(body, {}, function (error: any, success: any) {
+        forSubscriptions(pgClient, 'newHeads', function (row: any){
+          sendPayload(expressWsApp, row.ws_key, row.sub_id, error || success)
+        })
+      })
+    }
+
+    if (message.channel === 'log') {
+      const parsedObject = JSON.parse(message.payload)
+      const params = { "fromBlock": parsedObject.blockId, "toBlock": parsedObject.blockId, "index": parsedObject.index }
+      const body = {"jsonrpc": "2.0", "id": 1, "method": "eth_getLogs", "params": [params] }
+      jaysonWsServer.call(body, {}, function (error: any, success: any) {
+        forSubscriptions(pgClient, 'logs', function (row: any){
+          const address = (row.filter && row.filter.address && row.filter.address.id.toLowerCase()) || null
+          const topics = (row.filter && row.filter.topics) || null
+          if(address && address != success.result[0].address) {
+            // Skip delivery
+          } else if(topics && topics.filter(function(x: any) { return success.result[0].topics.includes(x) } ).length == 0) {
```

Comment:
Prefer [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) for short inline lambdas.